src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -122,23 +120,24 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
         this.setState({
             error: message,
             isLoading: false,
-            permissions: {},
             versions: [],
         });
     };
 
-    handleFetchSuccess = ([file, versions]): [BoxItem, FileVersions] => {
+    handleFetchSuccess = ([fileResponse, versionsResponse]): [BoxItem, FileVersions] => {
         const { api } = this.props;
-        const { entries } = api.getVersionsAPI(false).addCurrentVersion(versions, file) || {};
+        const versionsApi = api.getVersionsAPI(false);
+        const versionsWithCurrent = versionsApi.addCurrentVersion(versionsResponse, fileResponse);
+        const versionsWithPermissions = versionsApi.addPermissions(versionsWithCurrent, fileResponse);
+        const { entries: versions } = versionsApi.sortVersions(versionsWithPermissions) || {};
```

Comment:
I get the case for the other 2, but would it make sense to sort all versions entries all the time to prevent each caller from doing it?

Reply:
I figured it'd be best to leave it to the calling component due to the churn we had with sorting items in the activity feed.